src/cmd/ksh93/include/lexstates.h

```diff
@@ -23,6 +23,7 @@
 #include <wchar.h>
 #include <wctype.h>
 
+#define S_NOP 0    // absence of a state change, do nothing.
```

Comment:
I like the introduction of this symbol rather than using the magic `0` below but it should be used in the state tables like `sh_lexstate0` as well rather than magic zero in those tables.